apps/web/components/MegaMenu/MegaMenu.vue

```diff
@@ -39,7 +39,9 @@
           }
         "
       >
-        <li v-for="(menuNode, index) in categoryTree" :key="index">
+        <li v-if="categoryTree.length === 0" class="h-10"></li>
```

Comment:
**I'll try to explain bellow, as we have 2 scenarios:**
1. server-side rendering (on first-time visit, hard-reload)
    - wait to initialize required data (for server & client): ```app.vue -> setInitialDataSSR -> categoryTree```
    - after required data has been initialized, proceed with the rest of the app logic (which actually depends on initialization: session, user, basket, categories...)
2. client-side rendering (not first-time visit, no hard-reload)
    - because data has already being initialized and the client is aware of it's state, composables use existing state for their logic (unless a fresh data fetch is actually required)

Because of the above we find rendering fake (progressive) content in apps (eg Facebook), that's something we eventually have to consider imo...

**Nuxt error-prone scenario (missing expected data, hydration):**
1. server-side rendering (on first-time visit, hard-reload), eg for **userObj**
   - server fetch required composables data, without an actual await for **userObj** response
   - client composables data also initialized, and these depend on expected **userObj** data
   - because of the broken order of operations (**userObj** state from the server might not be available by the time client state needs/expects **userObj** to exist) app will crash, introduce data/hydration errors
2. client-side rendering (not first-time visit, no hard-reload), eg for **userObj**
    - client composables are trying to read their known version of **userObj** data
    - meanwhile server data might have changed, and composables are not aware of the changes
    
In a nutshell:

Data needs to initialize at the same time for sever&client.
When making a data state change, composables have to do it for server&client.
- this is critical for 'parent' composables, that have other composables depending on their data status 
- that means we need to set the value on server in something global (like a plugin or middleware), or defer it entirely to client with ```onMounted``` or ```onNuxtReady```